subprojects/core/src/main/java/org/gradle/internal/service/scopes/ProjectScopeServices.java

```diff
@@ -120,7 +121,7 @@ public ProjectScopeServices(final ServiceRegistry parent, final ProjectInternal
                 pluginServiceRegistry.registerProjectServices(registration);
             }
         });
-        addProvider(new WorkerSharedProjectScopeServices(project.getProjectDir()));
+        addProvider(new WorkerSharedProjectScopeServices(project.getProjectDir(), parent.get(ProjectCacheDir.class).getDir()));
```

Comment:
🤔 I didn't try it, but do we not allow `ProjectCacheDir` to be injected in `createDecompressionCacheFactory` below? Can't we do that lookup from the parent in the same way the other services are available?

Reply:
This is what I worked through with Gary.  We can't just inject `ProjectCacheDir` into `WorkerSharedProjectScopeServices` - it causes errors like:

```
         > Cannot create service of type FileSystemOperations using method WorkerSharedProjectScopeServices.createFileSystemOperations() as there is a problem with parameter #3 of type FileOperations.
            > Cannot create service of type DefaultFileOperations using method WorkerSharedProjectScopeServices.createFileOperations() as there is a problem with parameter #14 of type DecompressionCacheFactory.
               > Cannot create service of type DecompressionCacheFactory using method WorkerSharedProjectScopeServices.createDecompressionCacheFactory() as required service of type ProjectCacheDir for parameter #3 is not available.
```

when running `WorkerDaemonWorkerExecutorSampleIntegrationTest`, for instance.